src/node/in_memory_ext.rs

```diff
@@ -29,18 +29,11 @@ impl<S: ForkSource + std::fmt::Debug + Clone + Send + Sync + 'static> InMemoryNo
     /// # Returns
     /// The applied time delta to `current_timestamp` value for the InMemoryNodeInner.
     pub fn increase_time(&self, time_delta_seconds: u64) -> Result<u64> {
-        self.get_inner()
-            .write()
-            .map_err(|err| anyhow!("failed acquiring lock: {:?}", err))
-            .map(|mut writer| {
-                if time_delta_seconds == 0 {
-                    return time_delta_seconds;
-                }
-
-                let time_delta = time_delta_seconds.saturating_mul(1000);
-                writer.current_timestamp = writer.current_timestamp.saturating_add(time_delta);
-                time_delta_seconds
-            })
+        if time_delta_seconds == 0 {
+            return Ok(time_delta_seconds);
+        }
```

Comment:
Do we need to handle it as a special case? I think it's fine to handle it as a part of the normal flow.